### PR TITLE
docs: add Avatar component docs

### PR DESCRIPTION
**Pull request by didoesdigital**
Closes: https://github.com/cultureamp/kaizen-design-system/issues/225

### CODE REVIEW

File: site/docs/components/avatar.mdx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+<Dont>
+
+- Avoid an avatar when most people or groups won't have an image to show.
+
+</Dont>
+</DosAndDonts>
+
+## See also
+
+- [Link vs button](/guidelines/link-vs-button/).
```
Comment (ActuallyACat) [inline site/docs/components/avatar.mdx]: /meganit Are these two guidelines related?

Comment (didoesdigital) [reply site/docs/components/avatar.mdx]: If an avatar is linked to a user profile page, there would be temptation to implement it as a button because it looks more buttony styled than link-ish.
